code/datums/components/footstep.dm

```diff
@@ -94,11 +94,15 @@
 	var/feetCover = (H.wear_suit && (H.wear_suit.body_parts_covered & FEET)) || (H.w_uniform && (H.w_uniform.body_parts_covered & FEET))
 
 	if(H.shoes || feetCover) //are we wearing shoes
+		if(HAS_TRAIT(H, TRAIT_SILENT_FOOTSTEPS))
```

Comment:
You could make these two checks as single one at the start of the proc instead.
Something like:
```
/datum/component/footstep/proc/play_humanstep()
  if(HAS_TRAIT(parent, TRAIT_SILENT_FOOTSTEPS))
    return
```